modules/artifact-activation/src/main/java/org/mule/runtime/module/artifact/activation/internal/nativelib/ArtifactCopyNativeLibraryFinder.java

```diff
@@ -115,24 +113,29 @@ private File copyNativeLibrary(String name, String libraryPath) {
 
   private String findLibraryLocally(String name) {
     String nativeLibName = System.mapLibraryName(name);
-    String extension = FilenameUtils.getExtension(nativeLibName);
-    Optional<URL> nativeLib =
-        stream(urls).filter((URL url) -> {
-          String fullPath;
-          try {
-            fullPath = url.toURI().toString();
-          } catch (URISyntaxException e) {
-            return false;
-          }
-          String baseName = getBaseName(fullPath);
-          return baseName.contains(name) && (fullPath.endsWith(extension) || IS_OS_MAC && fullPath.endsWith(JNILIB_EXTENSION));
-        }).findFirst();
-
-    if (nativeLib.isPresent()) {
-      return nativeLib.get().getFile();
-    } else {
-      return null;
+    String extension = getExtension(nativeLibName);
+    String localLibraryEqualName = null;
```

Comment:
Let's try to add some meaning to this. Like two methods called strictMatching and looseMatching. I think it will help you structure the code below too.